src/Bicep.Cli/Commands/BuildParamsCommand.cs

```diff
@@ -105,5 +157,25 @@ public async Task<int> RunAsync(BuildParamsArguments args)
         private bool IsBicepFile(string inputPath) => PathHelper.HasBicepExtension(PathHelper.FilePathToFileUrl(inputPath));
 
         private bool IsBicepparamsFile(string inputPath) => PathHelper.HasBicepparamsExension(PathHelper.FilePathToFileUrl(inputPath));
+
+        private static readonly ImmutableHashSet<JTokenType> SupportedJsonTokenTypes = new[] { JTokenType.Object, JTokenType.Array, JTokenType.String, JTokenType.Integer, JTokenType.Float, JTokenType.Boolean, JTokenType.Null }.ToImmutableHashSet();
+
+        private static SyntaxBase ConvertJsonToBicepSyntax(JToken token) =>
+        token switch
+        {
+            JObject @object => SyntaxFactory.CreateObject(@object.Properties().Where(x => SupportedJsonTokenTypes.Contains(x.Value.Type)).Select(x => SyntaxFactory.CreateObjectProperty(x.Name, ConvertJsonToBicepSyntax(x.Value)))),
+            JArray @array => SyntaxFactory.CreateArray(@array.Where(x => SupportedJsonTokenTypes.Contains(x.Type)).Select(ConvertJsonToBicepSyntax)),
+            JValue value => value.Type switch
+            {
+                JTokenType.String => SyntaxFactory.CreateStringLiteral(value.ToString(CultureInfo.InvariantCulture)),
+                JTokenType.Integer => SyntaxFactory.CreatePositiveOrNegativeInteger(value.Value<long>()),
+                // Floats are currently not supported in Bicep, so fall back to the default behavior of "any"
```

Comment:
>Floats [](http://example.com/codeflow?start=19&length=6)

When we eventually add Bicep support, will this behavior change (which could be a breaking change) or are we going to leave it as is?

Reply:
We will likely have to change this behavior since we need to assign types that are as accurate as possible for correct type validation. 

This behavior is similar to that of `ConvertJsonToBicepType` from [SystemNamesapceType](https://github.com/Azure/bicep/blob/a32bf93c03f424441b78749018ba65068ef9f464/src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs#L1284) so I am thinking the breaking change (with the introduction of floats) would be in all locations where we are going from JSON to Bicep 

Alternatively, we could also just block usage of floats to avoid breaking change at least in `.bicepparam` files